From 9b50396a584a8b949e174bb99d019d6285a34724 Mon Sep 17 00:00:00 2001
From: Gu, Wangyi <wangyi.gu@intel.com>
Date: Fri, 17 May 2013 14:14:15 +0800
Subject: correct the offset address to avoid TOMBSTONE

BZ: 107198

It's android framework defect.
Root cause: memcpy access the invalid data scope of streamDescPtr->
decoderSpecificInfo for the offset address calculation is wrong.

Category:aosp improvement
Domain: Video.Media-video editor
Origin: internal
Upstream-Candidate: Yes

Change-Id: Iccc89b0262766429ce8e46d296737dcdd5c1dc66
Orig-Change-Id: Iae3cb5773802b26edef178dc64d3cc8105f8cb1b
Signed-off-by: Gu, Wangyi <wangyi.gu@intel.com>
Signed-off-by: chuzhoux <chux.zhou@intel.com>
---
 libvideoeditor/vss/3gpwriter/src/M4MP4W_Writer.c |    8 ++++----
 1 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/libvideoeditor/vss/3gpwriter/src/M4MP4W_Writer.c b/libvideoeditor/vss/3gpwriter/src/M4MP4W_Writer.c
index cdfc441..937d344 100755
--- a/libvideoeditor/vss/3gpwriter/src/M4MP4W_Writer.c
+++ b/libvideoeditor/vss/3gpwriter/src/M4MP4W_Writer.c
@@ -1144,22 +1144,22 @@ M4OSA_ERR M4MP4W_addStream(M4OSA_Context context,
                         PPSLength = DSI[10];
                         memcpy(
                             (void *)mMp4FileDataPtr->videoTrackPtr->DSI,
-                            (void *)((streamDescPtr->
+                            (void *)((M4OSA_UInt8 *)(streamDescPtr->
                             decoderSpecificInfo)+12), 2);
                         memcpy(
                             (void *)((mMp4FileDataPtr->videoTrackPtr->
-                            DSI)+2), (void *)((streamDescPtr->
+                            DSI)+2), (void *)((M4OSA_UInt8 *)(streamDescPtr->
                             decoderSpecificInfo)+28), SPSLength);
 
                         memcpy(
                             (void *)((mMp4FileDataPtr->videoTrackPtr->
                             DSI)+2 + SPSLength),
-                            (void *)((streamDescPtr->
+                            (void *)((M4OSA_UInt8 *)(streamDescPtr->
                             decoderSpecificInfo)+20), 2);
                         memcpy(
                             (void *)((mMp4FileDataPtr->videoTrackPtr->
                             DSI)+4 + SPSLength),
-                            (void *)((streamDescPtr->
+                            (void *)((M4OSA_UInt8 *)(streamDescPtr->
                             decoderSpecificInfo)+28 + SPSLength),
                             PPSLength);
                         /* - H.264 trimming */
-- 
1.7.4.1

